platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/Address.java

```diff
@@ -924,4 +877,29 @@ public String toString() {
                 + agreePublicKey + ", memo='"
                 + memo + '\'' + '}';
     }
+
+    /**
+     * Translates this address to the string format used in config.txt.
+     *
+     * @return the string form of this address that would appear in config.txt.
+     */
+    public String toConfigText() {
+        return "address, " + nickname + ", " + selfName + ", " + stake + ", " + ip4AddressText(addressInternalIpv4)
+                + ", " + portInternalIpv4 + ", " + ip4AddressText(addressExternalIpv4) + ", " + portExternalIpv4;
+    }
+
+    /**
+     * Returns the string form of the ip address given if there are 4 bytes present, otherwise a standard string for the
+     * array is returned.
+     *
+     * @param ip the array of bytes to turn into an ip address string.
+     * @return a string representation of the input array of bytes,
+     */
+    private String ip4AddressText(byte[] ip) {
```

Comment:
this looks like it should be a static utility method

Reply:
Worse, there's already a state method in Address that does this work and I didn't look for before writing this one.  Cody requested the capability to write config compatible address books as a separate PR.   Once that is merged in, I'll back out of this code change.